include/engine/algorithm.hpp

```diff
@@ -0,0 +1,87 @@
+#ifndef OSRM_ENGINE_ALGORITHM_HPP
+#define OSRM_ENGINE_ALGORITHM_HPP
+
+#include <type_traits>
+
+namespace osrm
+{
+namespace engine
+{
+namespace algorithm
+{
+
+// Contraction Hiearchy
+struct CH final
+{
+};
+// Contraction Hiearchy with core
+struct CoreCH final
+{
+};
+}
+
+namespace algorithm_trais
```

Comment:
Hrm, I tried a few tricks here and all of them failed. This is hard to do for free functions with overloads and non-const ref parameters. :-/ Keeping the manual switches.